### PR TITLE
add tracking option id

### PR DESCRIPTION
**Pull request by nimmen**
add tracking option id

### CODE REVIEW

File: src/XeroPHP/Models/Accounting/TrackingCategory.php [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <?php
-
```
Comment (timacdonald) [inline src/XeroPHP/Models/Accounting/TrackingCategory.php]: Add new line back in

File: src/XeroPHP/Models/Accounting/TrackingCategory.php [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 namespace XeroPHP\Models\Accounting;
 
 use XeroPHP\Remote;
 use XeroPHP\Models\Accounting\TrackingCategory\TrackingOption;
 
 class TrackingCategory extends Remote\Model
 {
+
```
Comment (timacdonald) [inline src/XeroPHP/Models/Accounting/TrackingCategory.php]: Remove new line

File: src/XeroPHP/Models/Accounting/TrackingCategory.php [hunk 3]
```diff
@@ -236,7 +237,7 @@ public function getOptions()
     public function addOption(TrackingOption $value)
     {
         $this->propertyUpdated('Options', $value);
-        if (! isset($this->_data['Options'])) {
+        if (!isset($this->_data['Options'])) {
```
Comment (timacdonald) [inline src/XeroPHP/Models/Accounting/TrackingCategory.php]: restore space after NOT (!) operator

File: src/XeroPHP/Models/Accounting/TrackingCategory.php [hunk 4]
```diff
@@ -262,4 +263,24 @@ public function setOption($value)
         $this->_data['Option'] = $value;
         return $this;
     }
+
+    /**
+     * @return string $value
+     * Return tracking option id
```
Comment (timacdonald) [inline src/XeroPHP/Models/Accounting/TrackingCategory.php]: Think this docblock should just be adjusted to the return type.

```
/**
 * @return string
 */
```

**Comment by timacdonald** [comment]
I'm not seeing this value available on the root TrackingCategory in the docs: https://developer.xero.com/documentation/api/tracking-categories

Could you provide a screenshot of a response that contains this value on the root object and not in the Options array?

File: src/XeroPHP/Models/Accounting/TrackingCategory.php (hunk 4 above)
Comment (nimmen) [reply src/XeroPHP/Models/Accounting/TrackingCategory.php]: Can adjust if needed, but there are other examples, so tried to make this more consistent.

Existing case:
    /**
     * @return string $value
     * Returns selected option name
     */
    public function getOption()

**Comment by nimmen** [comment]
Added back whitespace formatting.
In our case, we always get Options as NULL, not sure if this may be api version related, but it seems redundant/not working, or might be related to client setup?
Sending screenshot with actual issue(before my changes), just pushing this upstream so we would not have problems later, when updating. After my changes everything works well for us now.

Untitled

**Comment by nimmen** [comment]
As you can see we get back tracking option name, but not tracking option id.
What else seems redundant/not working:
"TrackingCategoryName" which is "Name"
"TrackingOptionName" which is "Option"
"Options" which is always NULL in my test cases
"Status" ? don't think it's working

**Pull request closed without merging**